### ISSUE TITLE
Equality checks fail on Unity 2018?

### ISSUE THREAD

**Issue by wledfor2** [wledfor2/PlayHooky#1]
There seems to be an issue with equality checks in the Hook method on Unity 2018. More research is necessary on this:
```
MissingMethodException: Method not found: 'System.Reflection.MethodInfo.op_Equality'.

UnityEngine.GameObject:Internal_AddComponentWithType(Type)
UnityEngine.GameObject:AddComponent(Type)
UnityEngine.GameObject:AddComponent()
Example.ExampleComponent:Awake()
UnityEngine.GameObject:Internal_AddComponentWithType(Type)
UnityEngine.GameObject:AddComponent(Type)
UnityEngine.GameObject:AddComponent()
Example.DLLMain:OnLoad()
```

**Comment by wledfor2** [comment]
This seems to be an issue with Unity's .NET 4.6 implementation. Casting the MethodInfo objects to object works as a workaround.

**Comment by wledfor2** [comment]
Fixed in ea2624acfb65b2cad51e4a81fb840ff93529ac81
